spec/formatting.md

```diff
@@ -222,6 +222,74 @@ the following steps are taken:
 
    The form that resolved _operand_ and _option_ values take is implementation-defined.
 
+   A _local-declaration_ binds the resolved value of an _expression_
+   to a _variable_.
+   Thus, the output of one _function_ is potentially the _operand_
+   of another _function_,
+   or the value of one of the _options_ for another function.
+   For example, in
+   ```
+   .input {$n :number minIntegerDigits=3}
+   .local $n1 = {$n :number maxFractionDigits=3}
+   ```
+   the output of the first call to `:number`
+   is the input of the second call to `:number`.
+
+   Implementations that provide a means for defining custom functions
+   SHOULD provide a means for function implementations
+   to return values that contain enough information
+   (e.g. the resolved _operand_ and _option_ values
+   that the function was called with)
```

Comment:
As this is an example and therefore non-normative this is _ok_, but I'd consider it surprising for a function to provide its input operand & option values rather than the resolved values.

Consider, for example, `{42 :number}`. The resolved value of the operand there is something like the _string_ `'42'`. Does it really make sense for that to be considered the "value" of this expression? Or is it more appropriate to allow and expect for the `:number` implementation to parse that string into a number `42`, and to present that as the value in its output?

Reply:
That's a good point. I changed it to: "a representation of the resolved...", which hopefully leaves enough room to make it clear that your example can work that way.